src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2864,14 +2863,14 @@ private static bool HasNonConstantInitializer(ArrayBuilder<ImmutableArray<FieldO
                     return;
                 }
 
-                bool reportMisplacedGlobalCode = !globalCodeAllowed && !m.HasErrors;
+                bool reportMisplacedGlobalCode = !m.HasErrors;
```

Comment:
Why does the absence of member errors indicate these members are in a bad global context?